### PR TITLE
fixed WALLET_PROPERTIES_ST serialization

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
closes https://github.com/brave/brave-browser/issues/4115
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
  - [x] Android
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc [hunk 1]
```diff
@@ -929,22 +929,48 @@ void saveToJson(JsonWriter* writer, const WALLET_PROPERTIES_ST& data) {
   }
   writer->EndObject();
 
-  writer->String("parameters_choices");
+  writer->String("parameters");
+  writer->StartObject();
+  writer->String("adFree");
+  writer->StartObject();
+
+  writer->String("currency");
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc]: currency is not part of WALLET_PROPERTIES_ST data structure

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc [hunk 2]
```diff
@@ -912,11 +912,11 @@ void saveToJson(JsonWriter* writer, const WALLET_PROPERTIES_ST& data) {
   writer->String("altcurrency");
   writer->String(data.altcurrency_.c_str());
 
-  writer->String("probi_");
+  writer->String("probi");
   writer->String(data.probi_.c_str());
 
   writer->String("balance");
-  writer->Double(data.balance_);
+  writer->String(std::to_string(data.balance_).c_str());
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc]: I agree using a String is the correct way, however right now the solution to be consistent is to change `d.HasMember("balance") && d["balance"].IsString()` to `d.HasMember("balance") && d["balance"].IsDouble()`

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc [hunk 3]
```diff
@@ -929,22 +929,48 @@ void saveToJson(JsonWriter* writer, const WALLET_PROPERTIES_ST& data) {
...
   writer->EndObject();
 
-  writer->String("parameters_choices");
+  writer->String("parameters");
+  writer->StartObject();
+  writer->String("adFree");
+  writer->StartObject();
+
+  writer->String("currency");
+  writer->String("");
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc]: nit: currency is not part of the data structure?

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc [hunk 4]
```diff
@@ -929,22 +929,48 @@ void saveToJson(JsonWriter* writer, const WALLET_PROPERTIES_ST& data) {
...
+
+  writer->String("range");
+  writer->StartObject();
+  writer->String("BAT");
 
-  writer->String("parameters_range");
   writer->StartArray();
-  for (auto & choice : data.parameters_range_) {
-    writer->Double(choice);
+  for (auto & range : data.parameters_range_) {
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc]: nit: `const auto& range`

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc [hunk 5]
```diff
@@ -929,22 +929,48 @@ void saveToJson(JsonWriter* writer, const WALLET_PROPERTIES_ST& data) {
   }
   writer->EndObject();
 
-  writer->String("parameters_choices");
+  writer->String("parameters");
+  writer->StartObject();
+  writer->String("adFree");
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc]: Where does `adFree` come from as I see no reference to this anywhere?

Comment (SergeyZhukovsky) [reply vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc]: @tmancey check that for `adFree` https://github.com/brave/brave-core/pull/2228/files#diff-8e22b475f70e7eca00ac43fcfb993885R873

**Comment by tmancey** [comment]
LGMT to me other than comment to @NejcZdovc, also @SergeyZhukovsky can you tick the appropriate boxes in the description please, thanks

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc (hunk 2 above)
Comment (SergeyZhukovsky) [reply vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc]: I think we discussed on a call that we are going to change all double to string internally. I don't think we should change it there.

**Comment by SergeyZhukovsky** [comment]
I marked the checkboxes and commented on the above question to @NejcZdovc

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc (hunk 2 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc]: structure currently have it as string so we should leave it as string for now

**Pull request merged**